### ISSUE TITLE
Cannot build on Debian 8 (Jessie), ./build.sh fails

### ISSUE THREAD

**Issue by nmschulte** [OmniSharp/omnisharp-roslyn#651]
As the title states, running the build script on Debian 8 fails.

```
nmschulte@debian:~/src/omnisharp-roslyn$ git fetch --all --prune
Fetching origin
nmschulte@debian:~/src/omnisharp-roslyn$ git status
On branch dev
Your branch is up-to-date with 'origin/dev'.
nothing to commit, working directory clean
nmschulte@debian:~/src/omnisharp-roslyn$ git clean -dxf
Removing .tools/
nmschulte@debian:~/src/omnisharp-roslyn$ ./build.sh 
Downloading NuGet...
Feeds used:
  /home/nmschulte/.local/share/NuGet/Cache
  https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json
  https://api.nuget.org/v3/index.json

Restoring NuGet package Cake.0.10.1.
WARNING: Unable to find version '0.10.1' of package 'Cake'.
Restoring NuGet package Newtonsoft.Json.8.0.3.
WARNING: Unable to find version '8.0.3' of package 'Newtonsoft.Json'.
Could not find Cake.exe at '/home/nmschulte/src/omnisharp-roslyn/scripts/../.tools/Cake/Cake.exe'.
```

I was trying to build this on Debian Sid (unstable), but the .NET Core CLI tooling does not work properly with recent Linux kernels; it crashes, and the only known work around is to use older kernels, and hence my attempt to build on Debian Jessie (stable; v8).

I received this error while first trying on Debian unstable, but the issue was an incompatible version of NuGet (and then, the version of the `~/.nuget/packages/` format.or such; completely clearing NuGet and then running the script worked, except then that `dotnet restore` crashes...).

What can I do to help debug this?  Thank you for your help!

Also, I intend to use this with YouCompleteMe; is onmisharp-roslyn a drop-in replacement for omnisharp-server?

**Comment by nmschulte** [comment]
I replaced the `//.tools/nuget.exe` binary with the latest available from nuget.org, and now the output looks like this:

```
nmschulte@debian:~/src/omnisharp-roslyn$ ./build.sh 
Feeds used:
  /home/nmschulte/.local/share/NuGet/Cache
  https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json
  https://api.nuget.org/v3/index.json

Restoring NuGet package Cake.0.10.1.
WARNING: Unable to find version '0.10.1' of package 'Cake'.
  /home/nmschulte/.local/share/NuGet/Cache: Package 'Cake.0.10.1' is not found on source '/home/nmschulte/.local/share/NuGet/Cache'.
  https://api.nuget.org/v3/index.json: Unable to load the service index for source https://api.nuget.org/v3/index.json.
  This method must be implemented in derived classes
  https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json: Unable to load the service index for source https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json.
  This method must be implemented in derived classes

Restoring NuGet package Newtonsoft.Json.8.0.3.
WARNING: Unable to find version '8.0.3' of package 'Newtonsoft.Json'.
  /home/nmschulte/.local/share/NuGet/Cache: Package 'Newtonsoft.Json.8.0.3' is not found on source '/home/nmschulte/.local/share/NuGet/Cache'.
  https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json: Unable to load the service index for source https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json.
  This method must be implemented in derived classes
  https://api.nuget.org/v3/index.json: Unable to load the service index for source https://api.nuget.org/v3/index.json.
  This method must be implemented in derived classes

Could not find Cake.exe at '/home/nmschulte/src/omnisharp-roslyn/scripts/../.tools/Cake/Cake.exe'.
```

---
before replacing `nuget.exe`:

```
nmschulte@debian:~/src/omnisharp-roslyn$ ./.tools/nuget.exe
NuGet Version: 3.3.0.212
usage: NuGet <command> [args] [options] 
Type 'NuGet help <command>' for help on a specific command.
...
```
--
after replacing `nuget.exe`:
```
nmschulte@debian:~/src/omnisharp-roslyn$ ./.tools/nuget.exe
NuGet Version: 3.4.4.1321
usage: NuGet <command> [args] [options] 
Type 'NuGet help <command>' for help on a specific command.
...
```

**Comment by nmschulte** [comment]
Running on Debian 9 (testing, Stretch) seems to work just fine.

**Comment by DustinCampbell** [comment]
Building on Debian 8 should be fixed with #835
